### PR TITLE
OPS-1156 View Agreement Details (part 1)

### PR DESCRIPTION
**Pull request by stevtek**
## What changed

* a new page to view an Agreement with the first tab to view Agreement Details implemented
  * top summary sections includes cards for Total Budget Lines and Total Agreement Value
  * details section includes Agreement metadata and Notes (with mock data) and History (with mock data) 
* NOTE: the second tab for Budget Lines is not implement in this PR
* new links from the Agreement list (on the Agreement name) to the new Agreement details page

## Issue

- #1156
  - #1210
  - #1211 
  - #1239 
  - #1240 
  - #1241

## How to test

* view the Agreements list page
* follow links from the Agreement name to the detail page
* verify the details page
* create an Agreement with budget lines with date_needed in the next three year
* verify the details page includes the bar chart with those

### Agreements list page with links to view details
image

### Agreement details page
**_NOTE: This image shows bar chart with FY 2043, but only the next three fiscal years will be shown_**
image

### CODE REVIEW

File: frontend/src/components/Agreements/AgreementDetailsCards/AgreementValuesCard.jsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                        enableLabel={true}
+                        isInteractive={false}
+                        role="application"
+                        ariaLabel="Total Agreement Value by Fiscal Year"
+                        borderRadius={2}
+                        valueFormat=">-$,.2f"
+                    />
+                </div>
+            ) : (
+                <p>No budget lines in the next 3 FYs</p>
```
Comment (fpigeonjr) [inline frontend/src/components/Agreements/AgreementDetailsCards/AgreementValuesCard.jsx:85]: needs more emoji 😄

File: frontend/package.json [hunk 2]
```diff
@@ -8,6 +8,7 @@
         "@fortawesome/free-regular-svg-icons": "6.4.0",
         "@fortawesome/free-solid-svg-icons": "6.4.0",
         "@fortawesome/react-fontawesome": "0.2.0",
+        "@nivo/bar": "0.83.0",
```
Comment (fpigeonjr) [inline frontend/package.json:11]: more Nivo charts pls

**Review by fpigeonjr — DISMISSED** [review]
great job

File: frontend/src/components/Agreements/AgreementDetailsCards/AgreementValuesCard.jsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        return acc;
+    }, {});
+    const fyValues = Object.keys(fyValuesMap).map((fy) => ({ fiscalYear: fy, amount: fyValuesMap[fy] }));
+    const totalValue = fyValues.reduce((acc, cur) => acc + cur.amount, 0);
+    const currentFiscalYear = fiscalYearFromDate(new Date());
+    const nextThreeFyValues = fyValues.filter((fyVal) => {
+        return fyVal.fiscalYear >= currentFiscalYear && fyVal.fiscalYear < currentFiscalYear + 3;
+    });
+
+    const barChartColors = [
```
Comment (tdonaworth) [inline frontend/src/components/Agreements/AgreementDetailsCards/AgreementValuesCard.jsx:32]: Do we want to reference global colors instead or store these somewhere where they can be referenced easier?

File: frontend/src/components/UI/Modal/Modal.jsx [hunk 4]
```diff
@@ -1,7 +1,19 @@
 import { useEffect, useRef } from "react";
 import PropTypes from "prop-types";
-import Notification from "../Notification";
+import LogItem from "../LogItem/LogItem";
```
Comment (fpigeonjr) [inline frontend/src/components/UI/Modal/Modal.jsx:3]: @johndeange `Notification` is now the generic `LogItem` component.

Comment (tdonaworth) [reply frontend/src/components/UI/Modal/Modal.jsx:3]: Not a hard stop, but `LogItem` kinda feels like the wrong name 🤔

File: frontend/src/components/Agreements/AgreementDetailsCards/AgreementValuesCard.jsx (hunk 3 above)
Comment (fpigeonjr) [reply frontend/src/components/Agreements/AgreementDetailsCards/AgreementValuesCard.jsx:32]: great idea @tdonaworth

File: frontend/src/components/UI/Modal/Modal.jsx (hunk 4 above)
Comment (fpigeonjr) [reply frontend/src/components/UI/Modal/Modal.jsx:3]: We started with `ListItem` but felt too generic and landed on `LogItem` after talking it over with @kimschulke .
😄 Suggestions are welcome.

Comment (tdonaworth) [reply frontend/src/components/UI/Modal/Modal.jsx:3]: This is simply for the individual items (notifications) right? I also see it is directly embedded in `Modal` - should every implementation of `Modal` have a `LogItem`? Should this be a more specific `Modal` like `NotificationModal` or whatever naming scheme we fall back to? 

Might need a quick walkthrough on this change, and its use-case..

Comment (tdonaworth) [reply frontend/src/components/UI/Modal/Modal.jsx:3]: Resolved offline

File: frontend/src/components/Agreements/AgreementDetailsCards/AgreementValuesCard.jsx (hunk 3 above)
Comment (fpigeonjr) [reply frontend/src/components/Agreements/AgreementDetailsCards/AgreementValuesCard.jsx]: done

**Review by tdonaworth — APPROVED** [review]

**Review by xlorepdarkhelm — APPROVED** [review]

**Pull request merged**